packages/teams-js/src/internal/appHelpers.ts

```diff
@@ -70,8 +87,37 @@ export function notifyFailureHelper(apiVersiontag: string, appInitializationFail
   ]);
 }
 
-export function notifySuccessHelper(apiVersionTag: string): void {
-  sendMessageToParent(apiVersionTag, app.Messages.Success, [version]);
+function supportsNotifySuccessResponse(): boolean {
```

Comment:
Am I reading it right that we are treating  notifySuccess a capability? Is there a specific reason why?

Reply:
The problem is that the previous behavior for `notifySuccess` was a fire-and-forget callback, therefore the host didn't send any response back to teams-js after the callback finished processing. As such, if I did the change without checking if the host answers back, I will wait indefinitely for a response to the callback as the host will never send one. In this manner, there are some newer hosts that will get a response from the callback and some older ones which will not, and I need to make the function behave accordingly.

In discussions with Trevor, we decided the best way to do it is following the `runtime` object in teams-js. Normally this is used for capabilities but on this case, it makes sense to use it, as it changes the behavior depending on the host is running in.